tests/conftest.py

```diff
@@ -0,0 +1,671 @@
+            "3d3451090602200c7e07e384b3620ccd2f97b5c08f5893357c653edc2b8570f0"
+            "99d9ff34a0285c012102d82f3fa29d38297db8e1879010c27f27533439c868b1"
+            "cc6af27dd3d33b243dec""fdffffff"
+            "01"
+            "d7ee7c0100000000""19""76a9140ea263ff8b0da6e8d187de76f6a362beadab781188ac"
+            "e3691900"
+        ),
+        prevout_idx=0,
+        prevout_amount=[bytes.fromhex("d7ee7c0100000000")]
+    )
+
+    segwit_tx = TrustedInputTestData(
+        tx=bytes.fromhex(
+            "02000000""0001"
+            "02"
+            "daf4d7b97a62dd9933bd6977b5da9a3edb7c2d853678c9932108f1eb4d27b7a9""00000000""00""fdffffff"
+            "daf4d7b97a62dd9933bd6977b5da9a3edb7c2d853678c9932108f1eb4d27b7a9""01000000""00""fdffffff"
+            "01"
+            "01410f0000000000""16""0014e4d3a1ec51102902f6bbede1318047880c9c7680"
+            "024730440220495838c36533616d8cbd6474842459596f4f312dce5483fe6507"
+            "91c82e17221c02200660520a2584144915efa8519a72819091e5ed78c52689b2"
+            "4235182f17d96302012102ddf4af49ff0eae1d507cc50c86f903cd6aa0395f32"
+            "39759c440ea67556a3b91b"
+            "0247304402200090c2507517abc7a9cb32452aabc8d1c8a0aee75ce63618ccd9"
+            "01542415f2db02205bb1d22cb6e8173e91dc82780481ea55867b8e753c35424d"
+            "a664f1d2662ecb1301210254c54648226a45dd2ad79f736ebf7d5f0fc03b6f8f"
+            "0e6d4a61df4e531aaca431"
+            "a7011900"
+        ),
+        prevout_idx=0,
+        prevout_amount=[bytes.fromhex("01410f0000000000")]
+    )
+
+    segwit_tx_2_outputs = TrustedInputTestData(
+        tx=bytes.fromhex(
+            "02000000""0001"
+            "01"
+            "1541bf80c7b109c50032345d7b6ad6935d5868520477966448dc78ab8f493db1""00000000""17"
+            "160014d44d01d48f9a0d5dfa73dab21c30f7757aed846a""feffffff"
+            "02"
+            "9b3242bf01000000""17""a914ff31b9075c4ac9aee85668026c263bc93d016e5a87"
+            "1027000000000000""17""a9141e852ac84f8385d76441c584e41f445aaf1624ea87"
+            "0247304402206e54747dabff52f5c88230a3036125323e21c6c950719f671332"
+            "cdd0305620a302204a2f2a6474f155a316505e2224eeab6391d5e6daf22acd76"
+            "728bf74bc0b48e1a0121033c88f6ef44902190f859e4a6df23ecff4d86a2114b"
+            "d9cf56e4d9b65c68b8121d"
+            "1f7f1900"
+        ),
+        num_outputs=2,
+        prevout_amount=[bytes.fromhex(amount) for amount in ("9b3242bf01000000", "1027000000000000")]
+    )
+
+    return [standard_tx, segwit_tx, segwit_tx_2_outputs]
+
+
+# ----------------------- Test data for test_btc_rawtx_ljs.py -----------------------
+
+
+def ledgerjs_test_data() -> List[List[LedgerjsApdu]]:
+    # Test data below is extracted from ledgerjs repo, file "ledgerjs/packages/hw-app-btc/tests/Btc.test.js"
+    ljs_btc_get_wallet_public_key = [
+        LedgerjsApdu(   # GET PUBLIC KEY - on 44'/0'/0'/0 path
+            commands=["e040000011048000002c800000008000000000000000"],
```

Comment:
Future improvement idea (not for this PR):

I don't know if firmware guys are able to read APDUs with the naked eye, but it seems we need hand-written Python stubs for each command. Here's an example:

```py
DISPLAY_MODE = Literal["none", "address", "token"]
DERIVATION_SCHEME = Literal["P2PKH", "P2SH-P2WPKH", "P2WPKH"]

def get_wallet_public_key(display: DISPLAY_MODE, scheme: DERIVATION_SCHEME):
    CLA = 0xE0
    INS = 0x40
    
    def g(derivation_path: str):
        # do some magic to serialize derivation_path to APDU input

    return g
```

I learnt from @hcleonis that these stubs can be complex especially with the chunking logic (in `DeviceAppProxy` class), but we should be able to centralize them in these stubs. The idea is for the reader of the app-bitcoin docs, to not care about the nitty-gritty of writing APDUs.

I'm pretty sure such a thing will be useful not only in internal Ledger projects but also in Bitcoin's Hardware Wallet Interface (HWI). It could be done as a separate Python package.

cc: @Gbillou

Reply:
We are :-).
`LedgerjsApdu` is used to send already formatted APDUs extracted from a Ledgerjs log to the app. This was needed during the implementation of a fix, before the test framework was what it is now. Actually the framework came about because after using Ledgerjs-pre-formatted APDUs, I needed to test the fix with my own data. 
The abstration you propose is now taken care of by the `DeviceAppBtc` class, supported by low-level APDU comm implemented in the `DeviceAppProxy` class.

As of today, the `LedderjsApdu` class is only used to replay some `GetPublicKey Apdu` APDUs to the BTC app and some `GetVersion` APDUs to the Zcash app. Eventually It shall disappear in favor of higher-level API (to be) added to the `DeviceAppBtc` class. I didn't do it due to lack of time, but it should be very easy to do.

Reply:
Removal of `LedgerjsApdu` class in favor of new "stub" API on `deviceappbtc.py` marked as TODO in test README